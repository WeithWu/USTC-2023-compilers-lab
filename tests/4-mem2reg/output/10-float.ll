; ModuleID = 'cminus'
source_filename = "/home/weith/gits/USTC-2023-compilers-lab/tests/4-mem2reg/functional-cases/10-float.cminus"

declare i32 @input()

declare void @output(i32)

declare void @outputFloat(float)

declare void @neg_idx_except()

define i32 @main() {
label_entry0:
  %op0 = alloca float
  %op1 = alloca float
  %op2 = alloca float
  store float 0x3ff19999a0000000, float* %op0
  store float 0x3ff8000000000000, float* %op1
  store float 0x3ff3333340000000, float* %op2
  %op3 = fmul float 0x3ff19999a0000000, 0x3ff8000000000000
  %op4 = fadd float %op3, 0x3ff3333340000000
  call void @outputFloat(float %op4)
  ret i32 0
}
