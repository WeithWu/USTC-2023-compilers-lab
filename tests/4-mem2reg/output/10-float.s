	.text
	.globl main
	.type main, @function
main:
	st.d $ra, $sp, -8
	st.d $fp, $sp, -16
	addi.d $fp, $sp, 0
	addi.d $sp, $sp, -80
.main_label_entry0:
# %op0 = alloca float
	addi.d $t1, $fp, -28
	st.d $t1, $fp, -24
# %op1 = alloca float
	addi.d $t1, $fp, -44
	st.d $t1, $fp, -40
# %op2 = alloca float
	addi.d $t1, $fp, -60
	st.d $t1, $fp, -56
# store float 0x3ff19999a0000000, float* %op0
	ld.d $t0, $fp, -24
	lu12i.w $t8, 260300
	ori $t8, $t8, 3277
	movgr2fr.w $ft0, $t8
	fst.s $ft0, $t0, 0
# store float 0x3ff8000000000000, float* %op1
	ld.d $t0, $fp, -40
	lu12i.w $t8, 261120
	ori $t8, $t8, 0
	movgr2fr.w $ft0, $t8
	fst.s $ft0, $t0, 0
# store float 0x3ff3333340000000, float* %op2
	ld.d $t0, $fp, -56
	lu12i.w $t8, 260505
	ori $t8, $t8, 2458
	movgr2fr.w $ft0, $t8
	fst.s $ft0, $t0, 0
# %op3 = fmul float 0x3ff19999a0000000, 0x3ff8000000000000
	lu12i.w $t8, 260300
	ori $t8, $t8, 3277
	movgr2fr.w $ft0, $t8
	lu12i.w $t8, 261120
	ori $t8, $t8, 0
	movgr2fr.w $ft1, $t8
	fmul.s $ft2, $ft0, $ft1
	fst.s $ft2, $fp, -64
# %op4 = fadd float %op3, 0x3ff3333340000000
	fld.s $ft0, $fp, -64
	lu12i.w $t8, 260505
	ori $t8, $t8, 2458
	movgr2fr.w $ft1, $t8
	fadd.s $ft2, $ft0, $ft1
	fst.s $ft2, $fp, -68
# call void @outputFloat(float %op4)
	fld.s $fa0, $fp, -68
	bl outputFloat
# ret i32 0
	addi.w $a0, $zero, 0
	addi.d $sp, $sp, 80
	ld.d $ra, $sp, -8
	ld.d $fp, $sp, -16
	jr $ra
	addi.d $sp, $sp, 80
	ld.d $ra, $sp, -8
	ld.d $fp, $sp, -16
	jr $ra
