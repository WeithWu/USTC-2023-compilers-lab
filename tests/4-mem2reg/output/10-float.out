2.850000
0
